File: Embedded C/04_Bitwise Operators/Worksheet_2/8.c
#include <stdio.h>

#define SHIFT_RIGHT(num) ((num) >> 1 )
#define CHECK_LSB(num) ((num) & 1U )

int main() {
    unsigned int num;
    int bit_pos = 0;

    printf("Input number: ");
    scanf("%u", &num);
    printf("%d -> %b\n", num, num);
    
    while (CHECK_LSB(num) == 0) {
        num = SHIFT_RIGHT(num);
        bit_pos++;
    }
    

    printf("The highest SET bit is at %dth Pos\n", bit_pos);

    return 0;
}

File: Python/03 Conditional Statement/8.py
val = int(input())
if 0 <= val <= 63:
    print("Quadrant 1")
elif 64 <= val <= 127:
    print("Quadrant 2")
elif 128 <= val <= 191:
    print("Quadrant 3")
elif 192 <= val <= 255:
    print("Quadrant 4")